### PR TITLE
displayed the data from the fetchAll() api call on the landing page

### PR DESCRIPTION
**Pull request by vladciuca**
- Integrated Coingecko Api base in .env file
- Moved the fetchAll() api call in the CoinList component
- Added Loading & Api Error handling on fetch all page
- Display api data on landing page

- Added Search Component - No functionality
- Installed react-icons component
- Installed react-chartsjs-2

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,69 +1,44 @@
...
+                </StyledLink>
               </li>
             </ul>
-          </nav>
-          <Header />
-  
+            <Search />
+          </Nav>
+
           {/* A <Switch> looks through its children <Route>s and
```
Comment (redliz5808) [inline src/App.js]: Remove the comment on lines 35 and 36.

File: src/pages/CoinList/index.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    }
+  };
+  componentDidMount() {
+    this.getCoinList();
+  }
+  render() {
+    const hasCoinList = !this.state.isLoading && this.state.coinList;
+
+    return (
+      <>
```
Comment (yoloOnTheBattlefield) [inline src/pages/CoinList/index.js]: why not a wrapping div here?

File: src/pages/CoinList/index.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+          <Ul>
+            {this.state.coinList.map((coin) => {
+              const priceChange = coin.market_cap_change_percentage_24h < 0;
+              return (
+                <li key={coin.id}>
+                  <CoinRank>#{coin.market_cap_rank}</CoinRank>
+                  <CoinName>
+                    <StyledLink to={`coin/${coin.name}`}>
+                      <span>
+                        <Img src={coin.image} alt="" />
```
Comment (yoloOnTheBattlefield) [inline src/pages/CoinList/index.js]: alt?

File: src/pages/CoinList/index.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        )}
+        {hasCoinList && (
+          <Ul>
+            {this.state.coinList.map((coin) => {
+              const priceChange = coin.market_cap_change_percentage_24h < 0;
+              return (
+                <li key={coin.id}>
+                  <CoinRank>#{coin.market_cap_rank}</CoinRank>
+                  <CoinName>
+                    <StyledLink to={`coin/${coin.name}`}>
```
Comment (yoloOnTheBattlefield) [inline src/pages/CoinList/index.js]: /coin

File: src/pages/CoinList/CoinList.sytles.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    font-weight: bold;
+    color: #5b486a;
+  }
+  ${InfoText} {
+    color: #5b486a;
+  }
+`;
+
+export const CoinTotalSupply = styled.div`
+  display: felx;
```
Comment (redliz5808) [inline src/pages/CoinList/CoinList.sytles.js]: Should this be "display: flex"?

File: src/pages/CoinPage/index.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import React from "react";
+
+export default class CoinPage extends React.Component {
+  state = {};
+  render() {
+    console.log(this.props.match.params.name);
```
Comment (yoloOnTheBattlefield) [inline src/pages/CoinPage/index.js]: no console

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: src/pages/CoinPage/index.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import React from "react";
+
+export default class CoinPage extends React.Component {
+  state = {};
```
Comment (redliz5808) [inline src/pages/CoinPage/index.js:4]: It doesn't look like this component needs state and could be converted to a functional component (unless you have plans for it in the future - then just ignore this comment (: ).

File: src/pages/Dashboard/index.js [hunk 8]
```diff
@@ -1,4 +1,8 @@
 import React from "react";
-import axios from "axios";
 
-export const Dashboard = ()=>(<div>IM ON THE DASHBOARD!</div>)
\ No newline at end of file
+export default class Dashboard extends React.Component {
+  state = {};
```
Comment (redliz5808) [inline src/pages/Dashboard/index.js:4]: Same for this component...is state needed?

File: src/pages/CoinPage/index.js (hunk 7 above)
Comment (vladciuca) [reply src/pages/CoinPage/index.js:4]: It will definitely need state lol

File: src/pages/CoinList/index.js (hunk 3 above)
Comment (vladciuca) [reply src/pages/CoinList/index.js]: removed alt

File: src/pages/CoinPage/index.js (hunk 6 above)
Comment (vladciuca) [reply src/pages/CoinPage/index.js]: removed console.log

File: src/App.js (hunk 1 above)
Comment (vladciuca) [reply src/App.js]: removed the comments

File: src/pages/CoinList/CoinList.sytles.js (hunk 5 above)
Comment (vladciuca) [reply src/pages/CoinList/CoinList.sytles.js]: nice catch

File: src/App.js [hunk 9]
```diff
@@ -1,69 +1,42 @@
...
+            <Search />
+          </Nav>
+
+          <Switch>
             <Route exact path="/">
-              <Home list={this.state.list}/>
-            </Route>
-            <Route exact path="/coin">
-              <Coin />
+              <CoinList />
```
Comment (yoloOnTheBattlefield) [inline src/App.js:38]: I preffer you to use this pattern <Route exact path="/" component={CoinList}>

File: src/pages/CoinList/index.js [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+  CoinPrice,
+  CoinPriceChange,
+  CoinMarketCap,
+  CoinSupply,
+  CointCirculatingSupply,
+  CoinTotalSupply,
+  ChartBox,
+} from "./CoinList.sytles";
+import { FaCaretUp, FaCaretDown } from "react-icons/fa";
+import { CgInfinity } from "react-icons/cg";
```
Comment (yoloOnTheBattlefield) [inline src/pages/CoinList/index.js]: absolute imports before relative imports

File: src/pages/CoinList/index.js [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+    const hasCoinList = !this.state.isLoading && this.state.coinList;
+
+    return (
+      <div>
+        {this.state.isLoading && <div>Loading...</div>}
+        {this.state.hasError && (
+          <div>There was a problem fetching your data..</div>
+        )}
+        {hasCoinList && (
+          <Ul>
```
Comment (yoloOnTheBattlefield) [inline src/pages/CoinList/index.js]: Give this A different name

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

**Review by yoloOnTheBattlefield — APPROVED** [review]

**Pull request merged**